src/expr/src/scalar/func.rs

```diff
@@ -2726,9 +2733,9 @@ impl BinaryFunc {
             ))
             .nullable(true),
 
-            ListLengthMax { .. } | ArrayLength | ArrayLower | ArrayUpper => {
-                ScalarType::Int64.nullable(true)
-            }
+            ArrayLength | ArrayLower | ArrayUpper => ScalarType::Int64.nullable(true),
```

Comment:
Oh noo, these should all be Int32 too. This PR is ballooning in scope, so feel free to file a follow-up issue for this one instead.

Reply:
Ill merge as-is, and cleanup the array ones in a later pr!

Reply:
Sounds good. I filed #11527 to track!

Reply:
https://github.com/MaterializeInc/materialize/pull/11532